### PR TITLE
Add a static json endpoint for supporting deep linking on iOS

### PR DESCRIPTION
**Pull request by uzi**
Also a quick update to the robots.txt -- blacklisting
another crawler to match what's on www.reddit.com.

:eyeglasses: @schwers @prashtx

### CODE REVIEW

**Comment by uzi** [comment]
This is for https://reddit.atlassian.net/browse/MOBILEWEB-427

File: src/server/routes.es6.js [hunk 1]
```diff
@@ -3,6 +3,25 @@ import crypto from 'crypto';
 
 import constants from '../constants';
 
+// Static file needed to support deep links on iOS
+const appleAppSiteAssociationObj = {
+  'webcredentials': {
```
Comment (prashtx) [inline src/server/routes.es6.js]: 🔕 We don't need quotes around the keys here, but apparently our lint rules don't care, so I suppose it's not a big deal.

File: src/server/routes.es6.js [hunk 2]
```diff
@@ -36,6 +59,11 @@ const serverRoutes = function(app) {
     `;
   });
 
+  router.get('/apple-app-site-association', function * () {
+    this.body = appleAppSiteAssociation;
+    this.type = 'application/json';
```
Comment (prashtx) [inline src/server/routes.es6.js]: Will this add the `charset=utf-8` to the content type? If not, maybe it's better to set `this.body` to the object and let the rest of the stack take care of stringification/content-type/etc. Presumably that also gives us control over having pretty-print stringification in dev and concise stringification in production.

**Comment by prashtx** [comment]
💅

File: src/server/routes.es6.js (hunk 1 above)
Comment (schwers) [inline src/server/routes.es6.js]: 🔕 Style wise, we normally don't avoid quotes around object keys (e.g.)[https://github.com/reddit/reddit-mobile/blob/master/src/propTypes.es6.js]. I'll defer to you're thinking on this because it might be something we update often via copy-paste new versions, in which case keeping quotes is probably easier.

**Comment by uzi** [comment]
:haircut:

**Comment by prashtx** [comment]
🐟

**Comment by schwers** [comment]
:fish:

File: src/server/routes.es6.js [hunk 3]
```diff
@@ -13,6 +31,10 @@ const serverRoutes = function(app) {
       User-agent: 008
       Disallow: /
 
+      # 80legs' new crawler
```
Comment (kjoconnor) [inline src/server/routes.es6.js]: Can you make this a separate commit?

Comment (uzi) [inline src/server/routes.es6.js]: Sure
